src/libs/ajax/leonardo/models/runtime-config-models.ts

```diff
@@ -0,0 +1,70 @@
+export interface GpuConfig {
+  gpuType: string
+  numOfGpus: number
+}
+
+export type CloudService = 'GCE' | 'DATAPROC' | 'AZURE_VM'
+export const cloudServiceTypes: Record<CloudService, CloudService> = {
+  GCE: 'GCE',
+  DATAPROC: 'DATAPROC',
+  AZURE_VM: 'AZURE_VM'
+}
+
+export interface BaseRuntimeConfig {
+  cloudService: CloudService
+}
+
+export interface GceConfig extends BaseRuntimeConfig {
+  machineType: string
+  diskSize: number
+  bootDiskSize?: number //This is optional for supporting old runtimes which only have 1 disk. All new runtime will have a boot disk
+  zone: string
+  gpuConfig?: GpuConfig
+}
+
+export interface GceWithPdConfig extends BaseRuntimeConfig {
+  machineType: string
+  persistentDiskId: number
+  bootDiskSize: number
+  zone: string
+  gpuConfig?: GpuConfig
+}
+
+export interface DataprocConfig extends BaseRuntimeConfig {
+  numberOfWorkers: number
+  masterMachineType: string
+  masterDiskSize: number
+  workerMachineType?: string
+  workerDiskSize?: number
+  numberOfWorkerLocalSSDs?: number
+  numberOfPreemptibleWorkers?: number
+  properties: Record<string, string>
+  region: string
+  componentGatewayEnabled: boolean
+  workerPrivateAccess: boolean
+}
+
+export interface AzureConfig extends BaseRuntimeConfig {
+  machineType: string
+  persistentDiskId: number
+  region: string
+}
+
+export type GoogleRuntimeConfig = GceConfig | GceWithPdConfig | DataprocConfig
+export type RuntimeConfig = AzureConfig | GoogleRuntimeConfig
+
+//TODO: should really add a kind in the backend, WIP
```

Comment:
The `RuntimeConfig` models are serialized from 4 distinct scala models defined in the backend. Everything is fine and dandy when handling these pre-serialization in the backend because you can match trivially on the type in a strongly-typed language to sub-class it. We run into some trouble post-serialization because there is nothing we can do to re-type it once it leaves the JVM as JSON. We use a bit of a brittle approach here, and look for the presence of fields that exist only in a specific model's deserialization. Chatted a bit with @msilva-broad, and we are thinking it would be best to add a field (such as `configType`) in the backend, and add this info to the deserialization in the backend. 

Then, we can define `type RuntimeConfigType = 'azureConfig' | 'dataprocConfig' | ...` and `switch` at the callsite as detailed here https://www.fullstory.com/blog/discriminated-unions-and-exhaustiveness-checking-in-typescript/

This is pretty important to get right, as detailed in a later comment: